lib/ansible/module_utils/network/common/utils.py

```diff
@@ -275,7 +275,10 @@ def dict_merge(base, other):
             if key in other:
                 item = other.get(key)
                 if item is not None:
-                    combined[key] = dict_merge(value, other[key])
+                    if isinstance(other[key], dict):
+                        combined[key] = dict_merge(value, other[key])
```

Comment:
Shouldn't this be Mapping? 
``` python
from ansible.module_utils.common._collections_compat import Mapping
[...]
if isinstance(other[key], Mapping):
```

How deeply nested are your dicts?  If they're too deeply nested they'll run into the recursion limit.  We can cross that bridge if you start to get bug reports about it though....

Reply:
Will change it to mapping.

>  How deeply nested are your dicts?

It depends on how nested the data model is for a particular network device and the data model varies vastly from vendor to vendor so can't say exactly how deeply nested dicts can be.
`dict_diff` function for network use case is mainly used to get diff between the candidate configuration (user input or module arg spec) and the running config retrieved from the device when action `override` and `dict_merge` to merge candidate and running config when action is `merge`.

Reply:
Okay.  the default recursionlimit in python is 100 levels.  (But it might be set lower if someone has optimized their python build for low memory situations).  We'll just have to wait and see if we get any bug reports and fix this then.